### ISSUE TITLE
Connection's app name and icon are missing when installed in the work profile

### ISSUE THREAD

**Issue by cohesive-flight** [emanuele-f/PCAPdroid#217]
I have the app installed in my work profile running in root mode (RM), and it's not able to display the app's name and icon for any connection shown. Instead, it displays the app's UID and a blank image (with two small mountains) as its icon, despite the one installed in the main profile displaying both in RM just fine.

I'd like to suggest that you implement a way for PCAPdroid to be able to analyze connections from other profiles/users while installed in the main user/profile. This is mostly for convenience's sake and not a direct solution, as I'd like to just be able to view all connections my device is making from one app (though, this might just carry over the issue for any connections made in the work profile).

I'm using v1.5.0 of PCAPdroid on Android 12 (LineageOS for MicroG: 19.1-20220522).

**Comment by emanuele-f** [comment]
Additional note: another user reported this stack trace on 1.4.8, this is probably the root cause for this issues (1.5.0 adds a check for SecurityException which is probably masking the issue).

```
Build type: fdroid
Build version: 1.4.8 
Build date: 1981-01-01 01:01:02 
Current date: 2022-06-05 01:53:54 
Device: OnePlus LE2123 
OS version: Android 12 (SDK 31) 
 
Stack trace:  
java.lang.SecurityException: getPackagesForUid: UID 10458 requires android.permission.INTERACT_ACROSS_USERS_FULL or android.permission.INTERACT_ACROSS_USERS or android.permission.INTERACT_ACROSS_PROFILES to access user .
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2441)
	at android.os.Parcel.createException(Parcel.java:2425)
	at android.os.Parcel.readException(Parcel.java:2408)
	at android.os.Parcel.readException(Parcel.java:2350)
	at android.content.pm.IPackageManager$Stub$Proxy.getPackagesForUid(IPackageManager.java:5389)
	at android.app.ApplicationPackageManager$3.recompute(ApplicationPackageManager.java:1040)
	at android.app.ApplicationPackageManager$3.recompute(ApplicationPackageManager.java:1034)
	at android.app.PropertyInvalidatedCache.query(PropertyInvalidatedCache.java:562)
	at android.app.ApplicationPackageManager.getPackagesForUid(ApplicationPackageManager.java:1053)
	at com.emanuelef.remote_capture.AppsResolver.get(AppsResolver.java:103)
	at com.emanuelef.remote_capture.adapters.AppsStatsAdapter.lambda$setStats$0$com-emanuelef-remote_capture-adapters-AppsStatsAdapter(AppsStatsAdapter.java:151)
	at com.emanuelef.remote_capture.adapters.AppsStatsAdapter$$ExternalSyntheticLambda0.compare(Unknown Source:6)
	at java.util.TimSort.binarySort(TimSort.java:296)
	at java.util.TimSort.sort(TimSort.java:221)
	at java.util.Arrays.sort(Arrays.java:1492)
	at java.util.ArrayList.sort(ArrayList.java:1470)
	at java.util.Collections.sort(Collections.java:206)
	at com.emanuelef.remote_capture.adapters.AppsStatsAdapter.setStats(AppsStatsAdapter.java:150)
	at com.emanuelef.remote_capture.fragments.AppsFragment.doRefreshApps(AppsFragment.java:161)
	at com.emanuelef.remote_capture.fragments.AppsFragment.onViewCreated(AppsFragment.java:83)
	at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3019)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:551)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1758)
	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2849)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2784)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:262)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:478)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:246)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1470)
	at android.app.Activity.performStart(Activity.java:8154)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3799)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2341)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:233)
	at android.os.Looper.loop(Looper.java:344)
	at android.app.ActivityThread.main(ActivityThread.java:8184)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.pm.PackageManagerService$ComputerEngine.enforceCrossUserOrProfilePermission(PackageManagerService.java:4672)
	at com.android.server.pm.PackageManagerService$ComputerTracker.enforceCrossUserOrProfilePermission(PackageManagerService.java:5653)
	at com.android.server.pm.PackageManagerService.enforceCrossUserOrProfilePermission(PackageManagerService.java:12983)
	at com.android.server.pm.PackageManagerService.getPackagesForUid(PackageManagerService.java:10358)
	at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:2241)
```

**Comment by cohesive-flight** [comment]
I did have that same stack trace when viewing the connection tab in v1.4.8 when installed on the work profile. I was planning on making an issue about it yesterday, but I noticed a newer version was released on GitHub. Once I updated, it seemingly solved this issue, but looks like it only masked it instead.

**Comment by emanuele-f** [comment]
This should be addressed now with the [INTERACT_ACROSS_USERS permission](https://github.com/emanuele-f/PCAPdroid/commit/fec54e949935de92e246b9cdf097072c46fc26ea), which is now obtained when running the root capture for the first time. Please try [this signed pre-release apk](https://www.dropbox.com/s/cqn5pbnp1i9i6jy/PCAPdroid_1.5.1_rc1_gh.apk?dl=1) and let me know if it works for you

**Comment by cohesive-flight** [comment]
Apologies. It seems what I described with the WP only displaying connections from apps on the MP was always the case when the capture interface (CI) is set to its default (Internet). I've never configured the CI, so I didn't realize this, but I'm going to assume this occurs because the "Internet" CI might only refer to connections from the MP. Regardless, I noticed that only for the first ~10 seconds of starting the capture, it's able to retrieve connections from apps on the WP. 

When setting the CI to "All interfaces" for pre-release version in either the MP or WP, it does fix this issue. Thank you!

I'd like to note that when installed on the WP, the connections displayed from apps on the MP have the WP icon. This shouldn't matter much, as the one in the MP is able to distinguish the connections, but I'd like to just let you know.

**Comment by emanuele-f** [comment]
> "Internet" CI might only refer to connections from the MP

In PCAPdroid `Internet` refers to the interface that the system is using to route the traffic to the Internet. This allows you, for example, to continue capturing data when your device disconnects from wifi and uses the mobile connection, or vice-versa. Work profile should not affect this, so I'm not sure why you are experiencing this. You can monitor which interface PCAPdroid thinks is the current internet interface by opening the "Root log" in the left drawer, which reports "x is the Internet interface"

> I'd like to note that when installed on the WP, the connections displayed from apps on the MP have the WP icon

Thank you for reporting this, it's now fixed in e53f521ad37942c719b54b7aee2debe5dcffb648

> when installed on the WP and set at the default CI (Internet), it's able to retrieve connections from apps on the WP for only the first ~10 seconds of starting the capture

This is weird and should not happen. If you get other insights on this via the root log, please comment here

Closing this issue as solved
